lib/network/modules/components/nodes/util/ShapeBase.js

```diff
@@ -68,9 +68,33 @@ class ShapeBase extends NodeBase {
 
     if (this.options.label !== undefined) {
       // Need to call following here in order to ensure value for `this.labelModule.size.height`
-      this.labelModule.calculateLabelSize(ctx, selected, hover, x, y, 'hanging')
-      let yLabel = y + 0.5 * this.height + 0.5 * this.labelModule.size.height;
-      this.labelModule.draw(ctx, x, yLabel, selected, hover, 'hanging');
+      this.labelModule.calculateLabelDimensions(ctx, selected, hover, x, y, 'hanging'); 
+
+      let labelX = x;
+      let labelY = y;
+      let labelBaseline = 'middle';
+      const DISTANCE = 12; // note: 12 is maybe the line height??
+
+      switch(this.labelModule.position){ 
+        case 'inside': 
+          break; 
+        case 'left': 
+          this.labelModule.fontOptions.align = 'right'; // set font alignment to right 
```

Comment:
This should be done when the options are being processed in `setOptions`. Though a question comes to my mind whether it is necessary. If someone wants the text to be aligned away from the node then just let them do it.